### PR TITLE
Backpropagate changes on main/max_requirements into development

### PR DESCRIPTION
**Pull request by maxbriel**
This PR makes sure that any changes we did on the `max_requirement` or `main` branches that would affect the `development` branch files are backpropagated to it.

The majority is small documentation changes.
However, the 3 changes to discuss are:
1. `grids.io.py` file change from `dirname` to `mesa_dir` @mkruckow I believe you implemented this change last week. Should this be backpropagated to `developement`?
2. The requirements. Should we fix the requirement completely in `development` too?
3. visualisation of a single binary. @sgossage @astroJeff  You seem to have implemented this in v1 `main` should this be backpropagated into `development`?
https://github.com/POSYDON-code/POSYDON/blame/9bc93b5058434ef8b4dbfbe076e0162e04457bb9/posydon/visualization/VHdiagram.py#L52

### CODE REVIEW

**Comment by mkruckow** [comment]
The full, instead of the half change in `grids.io` is already part of PR #435 .

**Comment by mkruckow** [comment]
The changes of utils.data_download should update the tests introduced in PR #382 as well.

**Comment by astroJeff** [comment]
This update to the van den Heuvel diagram routines seems minor and reasonable. @sgossage, can you confirm?

**Comment by sgossage** [comment]
It looks like some of the surrounding code has changed for the VH diagrams in between now and older versions. 

I did some testing and sorry for the inconvenience, but I think we can get rid of most of the tweaks to the VH diagram code for single binary population files. 

Sending `index=0` as an argument should suffice, rather than having a boolean `binary` flag. With a `.h5` file that has only a single binary, sending anything other than `index=0` will throw an error, but perhaps that's reasonable and might be considered as expected behavior.

File: posydon/visualization/VH_diagram/SimulationModel.py [hunk 1]
```diff
@@ -14,11 +14,12 @@
 class SimulationModel:
     """Model containing the dataset of simulations."""
 
-    def __init__(self, filename, path="./"):
+    def __init__(self, filename, path="./", binary=False):
```
Comment (sgossage) [inline posydon/visualization/VH_diagram/SimulationModel.py]:
```suggestion
    def __init__(self, filename, path="./"):
```

File: posydon/visualization/VH_diagram/SimulationModel.py [hunk 2]
```diff
@@ -14,11 +14,12 @@
...
         """Initialize a SimulationModel instance."""
         self._df = None
         self.path = path
         self.filename = filename
+        self.binary = binary
```
Comment (sgossage) [inline posydon/visualization/VH_diagram/SimulationModel.py]:
```suggestion

```

File: posydon/visualization/VHdiagram.py [hunk 3]
```diff
@@ -39,6 +39,7 @@ def __init__(
         *,
         presentMode=PresenterMode.DETAILED,
         displayMode=DisplayMode.WINDOW,
+        binary = False,
```
Comment (sgossage) [inline posydon/visualization/VHdiagram.py]:
```suggestion
        binary = False,
```

File: posydon/visualization/VHdiagram.py [hunk 4]
```diff
@@ -48,7 +49,7 @@ def __init__(
         if not self._app:  # if not, create it
             self._app = QApplication([])
 
-        self._presenter = Presenter(filename=filename, path=path)
+        self._presenter = Presenter(filename=filename, path=path, binary=binary)
```
Comment (sgossage) [inline posydon/visualization/VHdiagram.py]:
```suggestion
        self._presenter = Presenter(filename=filename, path=path)
```

File: posydon/visualization/VH_diagram/Presenter.py [hunk 5]
```diff
@@ -328,9 +328,9 @@ class Presenter:
         "S2_profile": "",
     }
 
-    def __init__(self, filename, path="./"):
+    def __init__(self, filename, path="./", binary=False):
         """Initialize a Presenter instance."""
-        self._model = SimulationModel(filename=filename, path=path)
+        self._model = SimulationModel(filename=filename, path=path, binary=binary)
```
Comment (sgossage) [inline posydon/visualization/VH_diagram/Presenter.py]:
```suggestion
        self._model = SimulationModel(filename=filename, path=path)
```

File: posydon/visualization/VH_diagram/Presenter.py [hunk 6]
```diff
@@ -328,9 +328,9 @@ class Presenter:
         "S2_profile": "",
     }
 
-    def __init__(self, filename, path="./"):
+    def __init__(self, filename, path="./", binary=False):
```
Comment (sgossage) [inline posydon/visualization/VH_diagram/Presenter.py]:
```suggestion
    def __init__(self, filename, path="./"):
```

**Comment by maxbriel** [comment]
@sgossage I've removed the code, which also removed it from PR #451, since it won't need to be in the release then.

@mkruckow You don't expect an issue with including half the changes already?
For the `data_download` tests, would you like me to make a PR to your `matthias_tests` and include the updated checksum tests there?

**Comment by mkruckow** [comment]
I don't see a good reason to put in the half change, it will only create a merge conflict for the other PR, which includes the full change.
For the `data_download` stuff, I'd expect to finish the other longer pending PR first and then add the needed adoption for the new download details here (I can do that after the other PR is merged.).

**Comment by mkruckow** [comment]
When the latest after the pre release is done please merge branch [matthias_update_data_download](https://github.com/POSYDON-code/POSYDON/tree/matthias_update_data_download) in here to solve the failed unit test.

**Review by sgossage — APPROVED** [review]

**Review by astroJeff — APPROVED** [review]

**Pull request merged**